### ISSUE TITLE
sqlfluff isn't taking options from diff-quality

### ISSUE THREAD

**Issue by jpers36** [sqlfluff/sqlfluff#1759]
I'm not sure whether this is an issue with diff-cover or with sqlfluff.

## Expected Behaviour

The following command excludes the listed rules.

```
diff-quality --violations sqlfluff --compare-branch origin/qa --fail-under 100 --options "--exclude-rules L001,L003,L004,L008,L016"
```

## Observed Behaviour

No matter what I pass in the --options string, both --exclude-rules and --rules, all linting issues are flagged.

## Steps to Reproduce

See above.

## Dialect

TSQL

## Version

Python: 3.9.7
SQLFluff: 0.7.1

**Comment by barrywhart** [comment]
I looked into this. The `diff-quality` tool has two mechanisms to gather quality info:
* Built-in "drivers" that read a pre-existing output file from a quality tool
* Plug-in "reporters" that run a quality tool directly

SQLFluff is implemented using the latter technique. Currently, `diff-quality` only supports the `--options` feature you mention for the former (for more detail, see the [`diff-quality` source](https://github.com/Bachmann1234/diff_cover/blob/main/diff_cover/diff_quality_tool.py#L319L346)). In order to implement support for `--options`, both `diff-quality` and SQLFluff would need to be enhanced.

Can you specify these options using a `.sqlfluff` file instead?

**Comment by jpers36** [comment]
@barrywhart yes, that's what I've fallen back on.  I was hoping to reserve .sqlfluff for rules that we want to exclude permanently, while tweaking the diff-quality command in our pipeline to exclude rules temporarily (until we're fully functional) but I can manage it all in .sqlfluff for now.

**Comment by barrywhart** [comment]
Great! To be clear, this is doable, just more work than you might think. I'll leave it open, and we'll get to it at some point.
